lightning/src/ln/peer_handler.rs

```diff
@@ -262,24 +276,22 @@ impl<Descriptor: SocketDescriptor, CM: Deref> PeerManager<Descriptor, CM> where
 	/// Panics if descriptor is duplicative with some other descriptor which has not yet has a
 	/// disconnect_event.
 	pub fn new_outbound_connection(&self, their_node_id: PublicKey, descriptor: Descriptor) -> Result<Vec<u8>, PeerHandleError> {
-		let mut peer_encryptor = PeerChannelEncryptor::new_outbound(their_node_id.clone(), self.get_ephemeral_key());
-		let res = peer_encryptor.get_act_one().to_vec();
-		let pending_read_buffer = [0; 50].to_vec(); // Noise act two is 50 bytes
+		let mut handshake = PeerHandshake::new_outbound(&self.our_node_secret, &their_node_id, &self.get_ephemeral_key());
+		let (act, ..) = handshake.process_act(&[]).unwrap();
+		let res = act.unwrap().serialize();
 
 		let mut peers = self.peers.lock().unwrap();
 		if peers.peers.insert(descriptor, Peer {
-			channel_encryptor: peer_encryptor,
+			encryptor: PeerState::Authenticating(handshake),
 			outbound: true,
-			their_node_id: None,
+			their_node_id: Some(their_node_id.clone()),
 			their_features: None,
 
 			pending_outbound_buffer: LinkedList::new(),
 			pending_outbound_buffer_first_msg_offset: 0,
 			awaiting_write_event: false,
 
-			pending_read_buffer: pending_read_buffer,
-			pending_read_buffer_pos: 0,
-			pending_read_is_header: false,
+			pending_read_buffer: Vec::new(),
```

Comment:
If we're gonna have a buffer in the Conduit (which makes sense - it knows how much to read because it knows how to decrypt), can we not just drop it from Peers here?

Reply:
Yeah, but not quite yet. The very core part of peer_handler's logic depends on `while peer.pending_read_buffer.len() > 0`. That is not something where we should have visibility into other objects, so I think we can come up with a much better approach than simply checking if the conduit's read buffer is empty.